include/sys/zio_impl.h

```diff
@@ -110,32 +120,33 @@ enum zio_stage {
 	ZIO_STAGE_ISSUE_ASYNC		= 1 << 4,	/* RWF-- */
 	ZIO_STAGE_WRITE_COMPRESS	= 1 << 5,	/* -W--- */
 
-	ZIO_STAGE_CHECKSUM_GENERATE	= 1 << 6,	/* -W--- */
+	ZIO_STAGE_ENCRYPT		= 1 << 6,	/* -W--- */
```

Comment:
The current implementation turns off encryption for dnode blocks. This is necessary because the dmu code needs to be able to work with dnodes whether or not their bonus buffers are encrypted. By leaving the dnode blocks uncompressed we can still parse around the bonus buffers.